Client/List.ts

```diff
@@ -38,4 +40,48 @@ export abstract class List<
 				: this.connection.post<Response[]>(`${this.folder}/searches${generatePagination(page, size, sort)}`, pattern))
 		)
 	}
+	async getNextPaginated<R>(
+		previous: Paginated<R> | undefined,
+		callback: (
+			page: number,
+			size: number,
+			sort?: string,
+			request?: Record<string, any>
+		) => Promise<model.ErrorResponse | { list: R[]; totalCount: number } | R[]>,
+		request?: Record<string, any>,
+		chosenPage?: number,
+		chosenSize?: number,
+		chosenSort?: string
+	) {
+		const sort = chosenSort
+		let page = chosenPage
+		let size = chosenSize
+		let result
+		if (previous) {
+			if (previous.hasNextPage() == false) {
+				return new Paginated([], previous.totalCount, previous.page, previous.size, true)
+			}
+
+			page = previous.nextPage()
+			size = previous.size
+		} else {
+			page = page ?? 0
+			size = size ?? 20
+		}
+
+		const response = await callback(page, size, sort, request)
+		if (model.ErrorResponse.is(response)) {
+			result = response
+		} else {
+			let totalCount: number | undefined
+			if (!Array.isArray(response)) {
+				totalCount = response.totalCount
+
+				result = new Paginated(response.list, totalCount, page, size, response.list?.length == 0)
+			} else {
+				result = new Paginated(response, totalCount, page, size, response.length == 0)
```

Comment:
tbh for all of this, why not do

let list = response.list ?? response
result = new Paginated(list, totalCount, page, size, condition)

Reply:
type system doesnt seem to be smart enough

Reply:
what's the problem?

Reply:
it says list is not existing on R[]